kedro/framework/cli/cli.py

```diff
@@ -65,10 +68,9 @@ def info():
     plugin_versions = {}
     plugin_entry_points = defaultdict(set)
     for plugin_entry_point, group in ENTRY_POINT_GROUPS.items():
-        for entry_point in pkg_resources.iter_entry_points(group=group):
+        for entry_point in importlib_metadata.entry_points().select(group=group):
```

Comment:
Could this also raise some kind of exception? Or is it just the `entry_point.load` that we should wrap in the `try`?

Reply:
Though I guess this is just `kedro info` rather than being triggered on every single `kedro` command like the other instances so it doesn't matter if there's uncaught exceptions anyway.

Reply:
This is indeed the solution I go with initially, but later I find out the API was not consistent. So technically only Python 3.10 can use the stdlib.

![image](https://user-images.githubusercontent.com/18221871/169412304-ad7fafde-116b-4064-af41-f9caa6df8489.png)

Reply:
It could fail, but like you said it should be just `kedro info`, unlike other plugin where the program still run without loading it. I think it is ok to leave it.

Reply:
Urghh, what a pain. I think what you did in https://github.com/kedro-org/kedro/commit/3c12bccce22127191c9e802ff0f85e0e3afeff8b was ok, although probably it's more usual to just repeat the conditional import in multiple files.

BUT if the standard library API is only right in Python 3.10 then let's just forget about it and go for `importlib_metadata` all the way like you're doing now 👍

Reply:
I have considered that, but it also means we have to copy the same block in multiple files, and also anywhere that use `mocker.patch`, which is quite hard to read.

For the context, I started this PR with `importlib_metadata`, so I wasn't aware of the inconsistent API between the Python version. In theory, if I avoid using the `select` API, it could be compatible with python3.8-3.10. But for both case, we still need to have the conditional import, so I would rather just stick with `importlib_metadata`, unless this extra dependencies is causing trouble.

Reply:
Yeah, that totally makes sense. I'm definitely happy to go with `importlib_metadata` throughout.